src/main/java/edu/kis/powp/jobs2d/command/CommandStatisticVisitor.java

```diff
@@ -0,0 +1,75 @@
+package edu.kis.powp.jobs2d.command;
+
+import java.util.Iterator;
+import java.util.Vector;
+
+public class CommandStatisticVisitor implements CommandVisitor {
+    private double totalLength = 0;
+    private Vector<Integer> previousCoordinates = null;
+    private Vector<Integer> currentCoordinates = null;
+    private int operateToCount = 0;
+
+    private int setPositionCount = 0;
```

Comment:
There is already CommandCounterVisitor which calculates those values (see PR #3 ). You should not duplicate this feature (you could extend it).